Clojure/tests/syntax_test_clojure.clj

```diff
@@ -0,0 +1,2007 @@
+; ^^^^^ invalid.illegal.clojure
+;      ^- invalid
+;       ^^^^^^ invalid.illegal.clojure
+;              ^^^^^^ invalid.illegal.clojure
+;                     ^ constant.other.symbol.reader-macro.clojure
+;                      ^^^^^ invalid.illegal.clojure
+  01234N +01234N -01234N '01234N
+; ^^^^^^ invalid.illegal.clojure
+;       ^- invalid
+;        ^^^^^^^ invalid.illegal.clojure
+;                ^^^^^^^ invalid.illegal.clojure
+;                        ^ constant.other.symbol.reader-macro.clojure
+  10-20 10+20 1234n 1234m 1234. 1234.M
+; ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- constant
+  10.0/20 10/20.0 10/+20 10/-20
+; ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- constant
+  10:20:30
+; ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- constant
+  1r000
+; ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- constant
+
+; ## Ignore
+
+  ; valid symbols
+  .1234 .1234M
+; ^^^^^^^^^^^^- constant
+
+
+
+; # Symbols
+
+  ! $ % & * - _ = + | < > . / ?
+  ++ --
+  blah
+  blah/blah
+  blah.blah
+  blah.blah/blah
+  blah.blah/blah.blah
+  blah/blah/blah
+  blah1000
+  blah1000.blah1000
+  *blah*
+  blah'blah'
+  blah'''blah'''
+  blah:blah:blah
+  blah#blah#
+  blah///blah
+
+; ## Breaks
+
+  blah,blah,blah
+;     ^ comment.punctuation.comma.clojure
+  blah;blah;blah
+;     ^ comment.line.clojure punctuation.definition.comment
+  blah`blah
+;     ^ constant.other.symbol.reader-macro.clojure
+  blah~blah
+;     ^ constant.other.symbol.reader-macro.clojure
+  blah@blah
+;     ^ constant.other.symbol.reader-macro.clojure
+  blah^blah
+;     ^ constant.other.symbol.reader-macro.clojure
+  blah\blah
+;     ^^^^^ constant.character.clojure
+
+; ## Unaffected
+
+  'blah 'blah:blah
+; ^ constant.other.symbol.reader-macro.clojure
+;  ^- constant.other.symbol.reader-macro.clojure
+  [blah blah blah]
+
+; ## Invalid
+
+  //
+  blah:
+  blah::blah
+  /blah
+  blah/
+
+
+
+; # Keywords
+
+  :! :$ :% :& :* :- :_ := :+ :| :< :> :. :/ :?
+; ^^ constant.language.keyword.clojure
+;   ^- constant
+;    ^^ constant.language.keyword.clojure
+;       ^^ constant.language.keyword.clojure
+;          ^^ constant.language.keyword.clojure
+;             ^^ constant.language.keyword.clojure
+;                ^^ constant.language.keyword.clojure
+;                   ^^ constant.language.keyword.clojure
+;                      ^^ constant.language.keyword.clojure
+;                         ^^ constant.language.keyword.clojure
+;                            ^^ constant.language.keyword.clojure
+;                               ^^ constant.language.keyword.clojure
+;                                  ^^ constant.language.keyword.clojure
+;                                     ^^ constant.language.keyword.clojure
+;                                        ^^ constant.language.keyword.clojure
+;                                           ^^ constant.language.keyword.clojure
+  :++ :--
+; ^^^ constant.language.keyword.clojure
+;    ^- constant
+;     ^^^ constant.language.keyword.clojure
+  :blah
```

Comment:
Is `:blah` really a built in keyword?

Reply:
Clojure doesn't have any built-in syntactic keywords as seen in other languages, since it's defined in terms of EDN data structures. The keyword data type is the same as Erlang or Ruby atoms. They're literals similar to strings, interned for performance and used as map keys.

Reply:
In that case they should be `constant.other`, not `constant.language`: https://www.sublimetext.com/docs/3/scope_naming.html#constant.